instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py

```diff
@@ -855,6 +886,113 @@ def test_uninstrument_new_client(self):
+                        transport,
+                        (
+                            SyncOpenTelemetryTransport,
+                            AsyncOpenTelemetryTransport,
+                        ),
+                    )
+            HTTPXClientInstrumentor().uninstrument()
+
+        def test_instrument_client_with_proxy(self):
+            proxy_mounts = {
+                "http://": self.create_proxy_transport(
+                    "http://localhost:8080"
+                ),
+                "https://": self.create_proxy_transport(
+                    "http://localhost:8080"
+                ),
+            }
+
+            client = self.create_client(mounts=proxy_mounts)
+            self.assertEqual(len(client._mounts), 2)
+            for transport in client._mounts.values():
+                with self.subTest(transport):
+                    self.assertIsInstance(
+                        transport,
+                        (httpx.HTTPTransport, httpx.AsyncHTTPTransport),
+                    )
+
+            HTTPXClientInstrumentor().instrument_client(client)
+            result = self.perform_request(self.URL, client=client)
+            self.assertEqual(result.text, "Hello!")
+            self.assert_span(num_spans=1)
+            self.assertEqual(len(client._mounts), 2)
+            for transport in client._mounts.values():
+                with self.subTest(transport):
+                    self.assertIsInstance(
+                        transport,
+                        (
+                            SyncOpenTelemetryTransport,
+                            AsyncOpenTelemetryTransport,
+                        ),
+                    )
+            HTTPXClientInstrumentor().uninstrument_client(client)
+
+        def test_uninstrument_client_with_proxy(self):
+            proxy_mounts = {
+                "http://": self.create_proxy_transport(
+                    "http://localhost:8080"
+                ),
+                "https://": self.create_proxy_transport(
+                    "http://localhost:8080"
+                ),
+            }
+            HTTPXClientInstrumentor().instrument()
+            client = self.create_client(mounts=proxy_mounts)
+            self.assertEqual(len(client._mounts), 2)
+            for transport in client._mounts.values():
+                with self.subTest(transport):
+                    self.assertIsInstance(
+                        transport,
+                        (
+                            SyncOpenTelemetryTransport,
+                            AsyncOpenTelemetryTransport,
+                        ),
+                    )
+
+            HTTPXClientInstrumentor().uninstrument_client(client)
+            result = self.perform_request(self.URL, client=client)
+            self.assertEqual(result.text, "Hello!")
+            self.assert_span(num_spans=0)
+            self.assertEqual(len(client._mounts), 2)
+            for transport in client._mounts.values():
```

Comment:
I think the logic in this for loop exist many times in the code, shall we create a helper function for it to avoid the repeat?